keras/engine/topology.py

```diff
@@ -2358,7 +2358,7 @@ def _updated_config(self):
 
         if hasattr(self, 'optimizer'):
             model_config['optimizer'] = self.optimizer.get_config()
-            model_config['loss'] = self.loss.__class__.__name__
+            model_config['loss'] = getattr(self.loss, '__name__', self.loss)
```

Comment:
if we were passed a function, we save its name. Else we assume we were passed a string.